Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -222,7 +222,9 @@ public final class PostgresConnection: @unchecked Sendable {
             promise: promise
         )
 
-        self.channel.write(HandlerTask.extendedQuery(context), promise: nil)
+        let writePromise = self.channel.eventLoop.makePromise(of: Void.self)
+        self.channel.write(HandlerTask.extendedQuery(context), promise: writePromise)
+        writePromise.futureResult.cascadeFailure(to: promise)
```

Comment:
Great catch! Can we please add for each of the following cases a `do {} catch {}` block in the following test:

```swift
    func testWeFailRequestsIfConnectionWasClosedBefore() async throws {
        let (connection, channel) = try await self.makeTestConnectionWithAsyncTestingChannel()

        try await connection.closeGracefully()

        XCTAssertEqual(channel.isActive, false)

        // 1. fail query
        do {
            _ = try await connection.query("SELECT version;", logger: self.logger)
            XCTFail("Expected to fail")
        } catch {
            print("\(error)")
        }

        // 2. fail listen
        // TODO!
    }
```

Add the test to `PostgresConnectionTests`

Reply:
done.